### PR TITLE
Support Python 3.13

### PR DESCRIPTION
**Pull request by Kludex**

### CODE REVIEW

File: starlette/middleware/gzip.py [hunk 1]
```diff
@@ -42,6 +42,7 @@ def __init__(self, app: ASGIApp, minimum_size: int, compresslevel: int = 9) -> N
     async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         self.send = send
         await self.app(scope, receive, self.send_with_gzip)
+        self.gzip_file.close()
```
Comment (Kludex) [inline starlette/middleware/gzip.py]: @graingert Is this good enough?

Comment (graingert) [reply starlette/middleware/gzip.py]: You might want to do it in a finally, or even better use `with`

```python
with BytesIO() as buffer, GzipFile(..., buffer) as gzip_file:
    await self.app(scope, receive, functools.partial(self.send_with_gzip, buffer, gzip_file))
```

**Comment by musicinmybrain** [comment]
I tried this PR as a patch for [`python-starlette`](https://src.fedoraproject.org/rpms/python-starlette) in Fedora Rawhide, and I can confirm that it appears to fix all of the Python 3.13 test regressions from #2614 and #2615.

I do like @graingert’s suggestion in https://github.com/encode/starlette/pull/2662#discussion_r1705310571, though.

**Review by adriangb — APPROVED** [review]
Approving pending https://github.com/encode/starlette/pull/2662#discussion_r1705310571 being applied

**Review by graingert — APPROVED** [review]

**Pull request merged**